gogoproto/helper.go

```diff
@@ -242,7 +250,7 @@ func EnabledGoStringer(file *google_protobuf.FileDescriptorProto, message *googl
 }
 
 func HasGoGetters(file *google_protobuf.FileDescriptorProto, message *google_protobuf.DescriptorProto) bool {
-	return proto.GetBoolExtension(message.Options, E_GoprotoGetters, proto.GetBoolExtension(file.Options, E_GoprotoGettersAll, true))
+	return proto.GetBoolExtension(message.Options, E_GoprotoGetters, proto.GetBoolExtension(file.Options, E_GoprotoGettersAll, true)) && !IsDropTypeDeclaration(message)
```

Comment:
The reason is that in `protoc-gen-gogo/generator/generator.go` the information needed to create the getters is not worked out unless the struct is created. I can try and refactor that part if you are ok with it.